migrations/versions/542bfbdef624_constrain_dependencies_to_have_no_.py

```diff
@@ -22,16 +22,8 @@
 def upgrade():
     # add NULLABLE 'id' column
     with op.batch_alter_table("deps_association_table", schema=None) as batch_op:
-        batch_op.add_column(sa.Column("id", sa.Integer(), nullable=True))
         batch_op.drop_constraint("deps_association_table_pkey", type_="primary")
-        batch_op.create_primary_key("deps_association_pk", columns=["id"])
-        batch_op.alter_column(
-            existing_type=sa.INTEGER(),
-            column_name="id",
-            autoincrement=True,
-            existing_autoincrement=True,
-            nullable=False,
-        )
+        batch_op.add_column(sa.Column("id", sa.Integer(), primary_key=True))
```

Comment:
Adding a non-nullable column without a default value or something for the existing rows typically fails. Did you test to on a db with  deps_associations in it?